### PR TITLE
Add backup and restore instructions

### PR DESCRIPTION
**Pull request by rocodes**
**Status** WIP/Ready for initial review
**Description** Document backup and restore process 
**Resolves** #46 
**Related** https://github.com/freedomofpress/securedrop-workstation/issues/120

The approach taken in this PR is to suggest backing up the vault vm and the configuration files from `dom0`, and nothing else. (See comms in linked issues above). 

I hope to improve on this approach eventually, because it's still a little unwieldy. Open questions include:
- Re-downloading and verifying the rpm and then shuttling it into dom0 seems like a real pain, but relying on a (potentially _very_ old) rpm + the updater also doesn't seem great. Currently, to be conservative, I recommend users re-download/verify/shuttle the RPM if it's not the current version. Open to feedback. 
- Currently, if users have configured custom RPC policies (such as tags that grant extended clipboard access), they have to manually re-add those after restoring the backup. Rationale: overwriting `/etc/qubes-rpc` in dom0 is risky, shouldn't be undertaken lightly, and has potential for breakage or bad consequences (for example, if VM names change and a hardcoded rpc policy is in place, or if a VM is granted access in an RPC policy but isn't restored in the backup, then if a future VM is created with that name, it could have unexpected permissions grandfathered in...yknow, stuff to be cautious/paranoid about). Again, tried to be conservative here, open to feedback.

I think it will make sense to have a much more streamlined backup/restore, but this is a start. (Future ideas: maybe a "bundle backup configs" button in the GUI,  maybe an easier way to grab/verify the RPM, and maybe a different consensus on whether or not we backup and restore more of the system, such as templates, so as to avoid long wall times/manual intervention during restore?)

### CODE REVIEW

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:63]: I've got light concerns about how little detail we're providing on the mechanics of creating a LUKS-encrypted volume, mounting it, and unlocking it. My personal backup procedures are roughly:

1. create `backup` appvm with no network
2. connect external hard drive , use sys-usb to connect it to `backup`
3. run cli `crytsetup luksOpen` and `mount` commands (bleh)
4. then switch back to dom0 and use the backup utility

I don't think we need to go into that much detail in our docs, it's much better to link out to upstream docs, as you've done above. It'd be an interesting exercise to ask someone without a ton of Qubes familiarity to run through these backup docs. Maybe we can try that at a later stage of review. =)

Comment (conorsch) [reply docs/admin/backup.rst:63]: I would expect a LUKS-encrypted drive to be automatically attached to `sd-devices` on fully provisioned Workstation. If that's the case, fine to use that to perform the backup!

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:92]: To reduce the number of "ifs", maybe we can just say explicitly here _don't_ install those default VMs. We don't care about personal, work, or untrusted—and vault will be coming from the backup.

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:46]: `s/sd-usb/sd-devices/`

File: docs/admin/backup.rst
Comment (conorsch) [reply docs/admin/backup.rst:63]: Ah, right: above it was referred to as `sd-usb`, and I'd planned to re-target that as `sys-usb`. Looks like `sd-devices` is what we want, and it's already addressed in a comment above.

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:129]: :rotating_light: See note in backup section about the potentially harmful recursive copy here—we only want two (2) files. It's most important to specify them explicitly by fullpath in the backup section, but I'd still do the same in the restore section, to guard against surprises.

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:152]: In the interest of reducing `if`s—which I find cause me to stop reading and think hard about whether the condition applies to my use case—let's just recommend installing from scratch again. It feels like an edge case for the back up and restore to happen on the same version of `securedrop-workstation-dom0-config`. Even in the case of a minor version bump, we want to make sure the new system uses the latest and greatest.

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:71]: Side note: we should be installing `diceware` in the SDW templates and instructing admins to use it.

File: docs/admin/backup.rst
Comment (conorsch) [reply docs/admin/backup.rst:12]: Similiarly, consider 

> a secure place to store backup credentials (such as a password manager on your primary laptop)

**Review by conorsch — COMMENTED** [review]
Very nice! Left extremely verbose comments throughout. Basically my feedback boils down to 

1. let's ignore most of what's in dom0
2. the fresh install pre-restore should be as pristine as possible, including latest version

Note that I haven't actually run through these instructions to verify the steps functionally. After you've had a chance to review the comments, please ping me again for a deeper review!

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst:58]: I agree with you that I don't personally restore dom0, but here are the things that I wonder about w.r.t dom0:
- ease of workflow for a user that has configured copy-paste between VMs or other custom RPC policies (very common eg with passwords and the vault vm)
- _if_ just doing a simple hardware migration for some reason, and a user already has the latest RPM in dom0, it seems rude to make them go through the whole 'download, verify, shuttle to dom0 again' procedure.

I'm in favour of rethinking where those files are stored. I think we can safely expect that they will live on in perpetuity wherever they are copied to, so we have to take that into account, esp with `sd-journalist.sec`.

Comment (conorsch) [reply docs/admin/backup.rst:58]: >  user that has configured copy-paste between VMs or other custom RPC policies, possibly with our assistance (very common eg with passwords and the vault vm)

Backing up dom0 won't help in that case. In order to support retaining clipboard access settings, we'd have to either 1) backup the `sd-*` VMs (which would include tag info for each VM); or 2) write a script to dump the tags from the SDW VMs, and then encourage the admin to re-apply the tags after restoring. Neither seem worthwhile to my eye, although at least mentioning the clipboard access tags in the backup/restore docs certainly seems reasonable.

> if just doing a simple hardware migration for some reason, and a user already has the latest RPM in dom0

That strikes me as a lot of conditionals. In order to reduce headaches for admins, I think our efforts would be best spent in trying to avoid the shuttle-to-dom0 workflow entirely, e.g. via `--enablerepo=qubes-templates-community`. That'll take a lot of work in coordination with Qubes upstream, so definitely a long-term goal. I defer to your judgment on what's most helpful to admins at present!

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst:33]: In my experience, tags are restored when a qube is restored. (I remembered saying this [a while back](https://github.com/freedomofpress/securedrop-workstation/issues/120#issuecomment-656918934), but I was doubting myself, so I just tried tagging and restoring a test qube and indeed the custom tags were still present).  

Backing up the RPC policy directories (with the idea that admins could audit them if need be, but not restore them directly) was some sort-of-futureproofing in case we are dealing with customized export workflows that involve customized qubes. But I can remove it and we can cross that bridge if/when we come to it.

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst]: Resolved in 05b8dfd, where I no longer suggest backing up `dom0`.

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst]: thanks for this flag-- fixed in 05b8dfd

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst]: I've removed all of this in 05b8dfd and left a note that advanced users may need to back up additional `dom0` components. I think for now, that's a reasonable tradeoff information-wise. If we have orgs using a lot of customized qubes and/or modifying RPC policies to accommodate their own workflows down the line, we can address this in more depth (or maybe we will develop a system that avoids the need for them doing this).

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst]: removed in 05b8dfd

File: docs/admin/backup.rst
Comment (rocodes) [reply docs/admin/backup.rst]: oh, I remember why I had the "if" and this workflow here -- if they clean install and then need to re-download the RPM, without the default application qubes, they don't actually have a network-connected VM with which to grab the RPM, so our instructions are technically incomplete.

Comment (rocodes) [reply docs/admin/backup.rst]: (fixed in c462517)

**Comment by rocodes** [comment]
@conorsch, thanks for your review! I've addressed your comments and now the backup/restore workflow involves:
- backup: copy `sd-journalist.sec` and `config.json` to the vault vm; back up vault vm

- restore: reinstall qubes, restore vault, re-create work VM, re-download RPM, copy in and verify config files from vault vm, run `apply`.

File: docs/admin/backup.rst
Comment (conorsch) [inline docs/admin/backup.rst:137]: Missing leading slash on `tmp`, should be `/tmp/`.

**Comment by conorsch** [comment]
Thanks for the copious edits, @rocodes! Looks great to my eye. Before merge, I'd like to see @creviera take a pass through it so it's got her formal approval. No further concerns from me.

**Comment by creviera** [comment]
on my radar: I can review today or tomorrow

File: docs/admin/backup.rst
Comment (creviera) [reply docs/admin/backup.rst:62]: confirmed this works as expected and documented

File: docs/admin/backup.rst
Comment (creviera) [reply docs/admin/backup.rst:70]: created followup issue

**Review by creviera — DISMISSED** [review]
lgtm, thanks ro!

**Review by creviera — APPROVED** [review]
looks great! :rocket:

**Pull request merged**
